### ISSUE TITLE
[Discussion] Workflow for large scale projects with multiple schemas

### ISSUE THREAD

**Issue by ganemone** [prisma/graphqlgen#249]
A common workflow for large scale projects is to write smaller self contained graphql schemas, and merge them into a single API via schema stitching. The current workflow for graphqlgen appears to be focused around generating code based on a single schema. I wanted to start a discussion around ideas for supporting generated types with a schema stitching workflow.

A naive approach could be to just run the existing code generation logic against multiple schemas. This probably would cover a large percentage of cases, but probably would not solve linking schemas with type extensions without some custom code. To accomplish this, we could update the config to support a list of options and/or alternatively update the CLI to support cli args rather than a file config.

Other ideas? Is supporting schema stitching something that is on your radar? I would be happy to collaborate on this.

**Comment by iamvishnusankar** [comment]
I did added this feature with a PR last day but got rejected. #238 , #242 

Type and interface linking would complex with multiple schema files. The best approach would be concatenating all scheme file and generating a single schema file before feeding into the build pipeline

From #242 
> Multiple schema files: This is actually already cover by graphql-import, which also handles lots possible errors by merging schemas.
Please refer #201
This has nothing to do with any schema validation or any other stuff. The only purpose of this is to concatenate multiple schema files into single file. This operation is conducted as part of the config orchestration.
Basically, it will find all the files mentioned in config.schema (wildcard/single file) and concatenate them to generate a single schema file. Once its done, config.schema path will get overrided by the newly created schema path before feeding into the build pipeline.

**Comment by ganemone** [comment]
> The best approach would be concatenating all scheme file and generating a single schema file before feeding into the build pipeline

Hm... I'm not sure I would take that approach. I would rather have separately generated types for each individual schema.

**Comment by schickling** [comment]
Schema stitching so far wasn't planned to be supported out-of-the-box. Could you elaborate a bit on your use case where running `graphqlgen` per sub-schema wouldn't be enough?

In #201, we're currently discussing how a set up could look like where you're splitting up a (single) schema across multiple files. Note however, that this seems to be another use case than was you described with multiple "self contained graphql schemas".
